File: src/main/java/me/soknight/minigram/chats/model/dto/client/ProfileDto.java
package me.soknight.minigram.chats.model.dto.client;

import com.fasterxml.jackson.annotation.JsonAlias;

import java.util.UUID;

public record ProfileDto(
        @JsonAlias({"id", "userId", "user_id"}) UUID userId,
        @JsonAlias({"name"}) String name,
        @JsonAlias({"photoUrl", "photo_url"}) String photoUrl
) {

}

File: src/main/kotlin/me/soknight/minigram/chats/dto/ProfileDto.kt
@file:OptIn(ExperimentalUuidApi::class)

package me.soknight.minigram.chats.dto

import kotlinx.serialization.Serializable
import me.soknight.minigram.chats.model.RelationStatus
import kotlin.uuid.ExperimentalUuidApi
import kotlin.uuid.Uuid

@Serializable
data class ProfileDto(
    val userId: Uuid,
    val name: String,
    val photoUrl: String? = null
)

@Serializable
data class ProfilePageDto(val count: Int, val data: List<ProfileDto>)

@Serializable
data class ProfileRelationDto(val status: RelationStatus, val profile: ProfileDto)
